lib/index.js

```diff
@@ -140,6 +140,14 @@ function getOptions(options) {
   options.sourceMap = getSourceMap(options);
   options.style = getStyle(options) || 0;
 
+  // context object represents node-sass environment
+  options.context = {
+    info: module.exports.info,
+    state: {},
+    sync: false,
```

Comment:
A good example of where these three items are useful is in scenarios where the custom importers (and eventually custom functions) come from npm modules. In these contexts, the importer function probably does not have any information about how it was invoked.

* state: This persistent object allows an importer or custom function to keep track of data across invocations. An example of this would be providing a custom importer that allows for import-once style functionality. I'm happy to remove this if we would prefer to let custom importer authors develop their own solutions
* sync: At the time of importer, the importer has no knowledge of if it was invoked via render or renderSync. While this is not as important for importers, it is definitely important for custom functions, as some custom functions may only work in a purely async manner. I'd really like to see this stay so that importer/function authors can know more about how they were invoked.
* info: It seemed valuable that an importer (or eventually custom function) could ask about the node-sass / libsass versions. I'm okay with removing this if we don't have a clear use case.